ingestr/main_test.py

```diff
@@ -2953,6 +2953,119 @@ def test_applovin_source(testcase):
     testcase()
 
 
+def frankfurter_test_cases() -> Iterable[Callable]:
+    def invalid_source_table(dest_uri):
+        schema = f"testschema_frankfurter_{get_random_string(5)}"
+        dest_table = f"{schema}.frankfurter_{get_random_string(5)}"
+        result = invoke_ingest_command(
+            "frankfurter://",
+            "invalid table",
+            dest_uri,
+            dest_table,
+        )
+        assert result.exit_code != 0
+        assert has_exception(result.exception, ValueError)
+
+    def interval_start_does_not_exceed_interval_end(dest_uri):
+        schema = f"testschema_frankfurter_{get_random_string(5)}"
+        dest_table = f"{schema}.frankfurter_{get_random_string(5)}"
+        result = invoke_ingest_command(
+            "frankfurter://",
+            "exchange_rates",
+            dest_uri,
+            dest_table,
+            interval_start="2025-04-11",
+            interval_end="2025-04-10",
+        )
+        assert result.exit_code != 0
+        assert has_exception(result.exception, ValueError)
+        assert "Interval-end cannot be before interval-start." in str(result.exception)
+
+    def interval_start_can_equal_interval_end(dest_uri):
+        schema = f"testschema_frankfurter_{get_random_string(5)}"
+        dest_table = f"{schema}.frankfurter_{get_random_string(5)}"
+        result = invoke_ingest_command(
+            "frankfurter://",
+            "exchange_rates",
+            dest_uri,
+            dest_table,
+            interval_start="2025-04-10",
+            interval_end="2025-04-10",
+        )
+        assert result.exit_code == 0
+    
+    def interval_start_does_not_exceed_current_date(dest_uri):
+        schema = f"testschema_frankfurter_{get_random_string(5)}"
+        dest_table = f"{schema}.frankfurter_{get_random_string(5)}"
+        start_date = pendulum.now().add(days=1).format("YYYY-MM-DD")
+        result = invoke_ingest_command(
+            "frankfurter://",
+            "exchange_rates",
+            dest_uri,
+            dest_table,
+            interval_start=start_date,
+        )
+        assert result.exit_code != 0
+        assert has_exception(result.exception, ValueError)
+        assert "Interval-start cannot be in the future." in str(result.exception)
+
+   
+    def interval_end_does_not_exceed_current_date(dest_uri):
+        schema = f"testschema_frankfurter_{get_random_string(5)}"
+        dest_table = f"{schema}.frankfurter_{get_random_string(5)}"
+        start_date = pendulum.now().subtract(days=1).format("YYYY-MM-DD")
+        end_date = pendulum.now().add(days=1).format("YYYY-MM-DD")
+        result = invoke_ingest_command(
+            "frankfurter://",
+            "exchange_rates",
+            dest_uri,
+            dest_table,
+            interval_start=start_date,
+            interval_end=end_date,
+        )
```

Comment:
Although I like the idea of these tests, it can cause subtle issues for our users.

For instance, depending on which timezone frankfurter uses, the `interval_end` parameter may be valid.

Reply:
solutions:
1. Remove --interval-end checks exceeding the current date. Users are responsible for providing the correct parameters.
    -- or --
2. Figure out what timezone frankfurter uses and translate all date/time inputs to that timezone and then run validation on end date.